packages/tinacms/src/unifiedClient/index.ts

```diff
@@ -47,18 +50,29 @@ export class TinaClient<GenQueries> {
     this.readonlyToken = token?.trim()
     this.queries = queries(this)
     this.errorPolicy = errorPolicy || 'throw'
+    this.cacheDir = cacheDir || ''
+  }
 
-    // if we're in node, use the fs cache
-    if (cacheDir && typeof require !== 'undefined') {
-      const { NodeCache } = require('tinacms/dist/cache')
-      this.cache = NodeCache(cacheDir, require('fs'))
+  async init() {
+    if (this.initialized) {
+      return
+    }
+    try {
+      if (this.cacheDir && typeof require !== 'undefined') {
```

Comment:
Why do we need to check if `require` is undefined here?

Reply:
Detects if nodejs environment which has fs/path/os apis (versus edge and browser runtime)